pl.c

```diff
@@ -921,13 +926,29 @@ const char *pl_marked_pl_name(void)
 	return pl_marked->name;
 }
 
-void pl_set_marked_pl_by_name(const char *name)
+struct playlist *pl_get_pl_by_name(const char *name)
 {
 	struct playlist *pl;
 	list_for_each_entry(pl, &pl_head, node) {
 		if (strcmp(pl->name, name) == 0) {
-			pl_marked = pl;
-			return;
+			return pl;
 		}
 	}
 }
+
+void pl_set_marked_pl_by_name(const char *name)
+{
+	pl_marked = pl_get_pl_by_name(name);
+}
+
+const char *pl_playing_pl_name(void)
+{
+	return pl_playing->name;
+}
+
+void pl_resume(const char *name, struct track_info *ti)
+{
+
+	struct playlist *pl = pl_get_pl_by_name(name);
+	// ...
```

Comment:
This is where I am struggeling. ATM I am trying `pl_play_track()`, but I get `window_set_sel: BUG: iter_is_empty(&win->top)`, so I guess I am missing some initializations.